metricbeat/module/linux/memory/data.go

```diff
@@ -129,21 +137,28 @@ func getHugePages(hostfs resolve.Resolver) (common.MapStr, error) {
 }
 
 // GetVMStat gets linux vmstat metrics
-func GetVMStat() (*sysinfotypes.VMStatInfo, error) {
-	// TODO: We may want to pull this code out of go-sysinfo.
-	// It's platform specific, and not used by anything else.
-	h, err := sysinfo.Host()
+func GetVMStat(hostfs resolve.Resolver) (map[string]uint64, error) {
+	vmstat_file := hostfs.ResolveHostFS("proc/vmstat")
+	content, err := ioutil.ReadFile(vmstat_file)
```

Comment:
`ioutil` has been deprecated on Go 1.16, I suggest using [`os.ReadFile`](https://pkg.go.dev/os#ReadFile) instead
```suggestion
	content, err := os.ReadFile(vmstat_file)
```